WixSharp/Services/Authorization/AuthorizationService.cs

```diff
@@ -0,0 +1,281 @@
+
+                var json = JToken.Parse(rawDataString);
+                return new WixAuthorizationResult(json.Value<string>("access_token"), json.Value<string>("refresh_token"));
+            }
+        }
+
+        /// <summary>
+        /// Authorizes an application installation, generating an access token for the given shop.
+        /// </summary>
+        /// <param name="code">The authorization code generated by Wix, which should be a parameter named 'code' on the request querystring.</param>
+        /// <param name="wixUrl">The store's *.myWixUrl.com URL, which should be a parameter named 'shop' on the request querystring.</param>
+        /// <param name="wixAppKey">Your app's public API key.</param>
+        /// <param name="wixAppSecretKey">Your app's secret key.</param>
+        /// <returns>The shop access token.</returns>
+        public static async Task<WixAuthorizationResult> Authorize(string code, string wixUrl, string wixAppKey, string wixAppSecretKey)
+        {
+            return (await AuthorizeWithResult(code, wixUrl, wixAppKey, wixAppSecretKey));
+        }
+
+        /// <summary>
+        /// Determines if an incoming request is authentic.
+        /// </summary>
+        /// <param name="querystring">A dictionary containing the keys and values from the request's querystring.</param>
+        /// <param name="wixAppSecretKey">Your app's secret key.</param>
+        /// <returns>A boolean indicating whether the request is authentic or not.</returns>
+        public static bool IsAuthenticRequest(IDictionary<string, string> querystring, string wixAppSecretKey)
+        {
+            var qs = querystring.Select(kvp => new KeyValuePair<string, StringValues>(kvp.Key, kvp.Value));
+
+            return IsAuthenticRequest(qs, wixAppSecretKey);
+        }
+
+        /// <summary>
+        /// Determines if an incoming request is authentic.
+        /// </summary>
+        /// <param name="querystring">The request's raw querystring.</param>
+        /// <param name="wixAppSecretKey">Your app's secret key.</param>
+        /// <returns>A boolean indicating whether the request is authentic or not.</returns>
+        public static bool IsAuthenticRequest(string querystring, string wixAppSecretKey)
+        {
+            return IsAuthenticRequest(ParseRawQuerystring(querystring), wixAppSecretKey);
+        }
+
+        /// <summary>
+        /// Determines if an incoming request is authentic.
+        /// </summary>
+        /// <param name="querystring">The collection of querystring parameters from the request. Hint: use Request.QueryString if you're calling this from an ASP.NET MVC controller.</param>
+        /// <param name="wixAppSecretKey">Your app's secret key.</param>
+        /// <returns>A boolean indicating whether the request is authentic or not.</returns>
+        public static bool IsAuthenticRequest(IEnumerable<KeyValuePair<string, StringValues>> querystring, string wixAppSecretKey)
```

Comment:
How do you check authenticity in WIX?

Reply:
When you receive a data payload from Wix, it will include a header called 'digest' - the header will hold a JSON web token (JWT) with the signed data. (Here's a sample decoded JWT.)

Use your app’s public key to verify that the data was sent by Wix.